gallery_ripper.py

```
@@ -1286,6 +1375,109 @@ def _yandex_page_diagnostics(html: str, response=None) -> str:
+def _yandex_fetch_search_html(search_urls, log=lambda msg: None):
+    """Fetch a Yandex result page, retrying alternate hosts and optional browser cookies."""
+    blocked = []
+    cookie_modes = [(None, "plain session")]
+
+    for mode_idx, (cookies, label) in enumerate(cookie_modes):
+        for search_url in search_urls:
+            headers = _yandex_headers_for(search_url)
+            try:
+                resp = session.get(search_url, headers=headers, cookies=cookies, timeout=25, allow_redirects=True)
+                resp.raise_for_status()
+            except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)